frontend/app/src/components/home/groupdetails/Members.svelte

```diff
@@ -105,27 +105,31 @@
 </div>
 
 {#if me !== undefined && me.memberKind === "full_member"}
-    <Member me={true} member={me} />
+    <Member 
+        me
+        member={me} 
+        canPromoteToOwner={me.role !== "owner" && client.isPlatformModerator()}
+        canDemoteToAdmin={me.role === "owner" && client.canChangeRoles(chat.chatId, me.role, "admin")}
+        canDemoteToMember={me.role !== "participant" && client.canChangeRoles(chat.chatId, me.role, "participant")}
```

Comment:
I don't _think_ you need the first part of the conditional here. Isn't it included within `canChangeRoles`?